pkg/infra/iac2/templates/secret_version/factory.ts

```diff
@@ -10,19 +10,21 @@ interface Args {
 }
 
 // noinspection JSUnusedLocalSymbols
-function create(args: Args): void {
-    if (fs.existsSync(args.Path)) {
-        new aws.secretsmanager.SecretVersion(
-            args.Name,
-            {
-                secretId: args.Secret.id,
-                //TMPL {{- if eq .Type.Raw "string" }}
-                secretString: fs.readFileSync(args.Path, 'utf-8').toString(),
-                //TMPL {{ else }}
-                //TMPL secretBinary: fs.readFileSync({{ .Path.Parse }}, 'base64').toString(),
-                //TMPL {{ end }}
-            },
-            { protect: args.protect }
-        )
-    }
+function create(args: Args): aws.secretsmanager.SecretVersion | undefined {
```

Comment:
Not sure if we need to reference the SecretVersion anywhere. This is mostly provided as a deploy-time convenience than it is for anything related to the topology.

Reply:
We don't currently, but I don't think that's a reason to specifically not create a variable in the generated  `index.ts` file.

Reply:
The reason it was as a void function (thus, not assignable to a variable) was because we'd have to sacrifice some readability to return the value from the factory function to make it assignable to a variable. Use-cases like these were why the void function support was added initially. IMO the `(() => {})()` pattern adds some cognitive overhead - one that I prefer to avoid if possible.